test/message-list.test.js

```diff
@@ -136,5 +137,147 @@ describe('message-list', () => {
       await nextRender(messageList);
       expect(messageList.scrollTop).to.be.equal(0);
     });
+
+    it('message list should set tab index on first item if new item list is shorter, and it does not have a item index corresponding to the previous item with tab index 0', async () => {
```

Comment:
There should be also a test for the case when the new list is not shorter, and the previous focused index is restored.

Reply:
I added:
- one test for when the list shortens, but still there is a matching index that can have tabindex=0
- another when the list grows.

Done.